cats/forecast.py

```diff
@@ -36,37 +36,97 @@ def __post_init__(self):
 
 class WindowedForecast:
 
-    def __init__(self, data: list[CarbonIntensityPointEstimate], window_size: int):
-        self.times = [point.datetime for point in data]
-        self.intensities = [point.value for point in data]
-        # Integration window size in number of time intervals covered
-        # by the window.
-        self.window_size = window_size
+    def __init__(
+            self,
+            data: list[CarbonIntensityPointEstimate],
+            duration: int,  # in minutes
+            start: datetime,
+    ):
+        self.data = data
+        self.data_stepsize = data[1].datetime - data[0].datetime
```

Comment:
This assumes that all the intervals are the same in the forecast, which may be a constraint down the line. And is it right that `bisect_left` below is there to avoid making this assumption? (otherwise, `ndata` could be calculated as `(end - start)/stepsize` or something like that).
It could be good to pick one (and if we assume all intervals are the same for now for simplicity, maybe worth leaving a TODO or comment to get back to it later).

Reply:
> This assumes that all the intervals are the same in the forecast

Yes

> it right that bisect_left below is there to avoid making this assumption? (otherwise, ndata could be calculated as (end - start)/stepsize or something like that).

This is how I originally implemented it, but here is a counter example. Say you've got 3 data points at 12:30, 13:00 and 13:30. Your job is 10 minutes long.  If you start at 12:40, you end at 12:50: that's `ndata =2`.  But if start time and end time fall on both side of a data point, say start at 12:55 and end at 13:05, then `ndata = 3`. Hope this helps? Sorry I wish we could draw on a board :)

Reply:
Ok, I've just added a a note #57 to remember to either fully check that or make it work for any interval sizes